lib/modules/nightMode.js

```diff
@@ -144,15 +154,26 @@ module.always = () => {
 module.go = () => {
 	handleAutomaticNightMode();
 
-	if (module.options.nightSwitch.value) {
+	if (module.options.nightSwitch.value && className() === 'res-nightmode') {
 		toggle.addMenuItem(i18n('nightModeToggleTitle'), '☽', '☀');
 	}
 };
 
 export function isNightModeOn(): boolean {
 	if (!Modules.isRunning(module)) return false;
+
+	const userCookie = getUserCookie();
+
 	// `toggle` contains the current nightMode state
-	return toggle ? toggle.enabled : module.options.nightModeOn.value;
+	// `userCookie` contains the decoded "USER" cookie data
+	return (toggle ? toggle.enabled : module.options.nightModeOn.value) || (userCookie.prefs.nightmode && isAppType("d2x"));
```

Comment:
`(userCookie && userCookie.prefs && userCookie.prefs.nightmode)` -- not guaranteed to have the `USER` cookie, especially before checking if `isAppType('d2x')`